checkpoint_schedules/basic_schedules.py

```diff
@@ -101,17 +101,22 @@ class SingleDiskStorageSchedule(CheckpointSchedule):
     move_data : bool
         Indicate whether the execution should move the data from
         `StorageType.DISK` to `StorageType.WORK`, rather than copy the data.
+    store_only_adj_deps : bool
+        Indicate whether only adjoint dependencies should be stored on disk.
+        This is useful when the forward solver identifies the data needed
+        for the adjoint-based gradient calculations.
 
     Notes
     -----
     Online, unlimited adjoint calculations permitted if `move_data` is `False`,
     one adjoint calculation permitted if `move_data` is `True`.
     """
 
-    def __init__(self, move_data=False):
+    def __init__(self, move_data=False, store_only_adj_deps=False):
```

Comment:
```suggestion
    def __init__(self, move_data=False, *, store_only_adj_deps=True):
```

Default `False` would be an API break, changing existing behaviour. New arguments should be keyword only.

Reply:
Yes. I just updated it.